install.sh

```diff
@@ -50,15 +50,61 @@ else
   # and ~/.linuxbrew (which is unsupported) if run interactively.
   HOMEBREW_PREFIX_DEFAULT="/home/linuxbrew/.linuxbrew"
   HOMEBREW_CACHE="${HOME}/.cache/Homebrew"
-  HOMEBREW_CORE_GIT_REMOTE="https://github.com/Homebrew/linuxbrew-core"
+  HOMEBREW_CORE_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/linuxbrew-core"
 
   STAT="stat --printf"
   CHOWN="/bin/chown"
   CHGRP="/bin/chgrp"
   GROUP="$(id -gn)"
   TOUCH="/bin/touch"
 fi
-HOMEBREW_BREW_GIT_REMOTE="https://github.com/Homebrew/brew"
+HOMEBREW_BREW_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/brew"
+
+# Parse remote URLs of Homebrew repositories from command line arguments if presented.
+parse_remote_urls_from_arguments() {
+  verify_git_remote_url() {
+    # Empty string or valid Git remote URL
+    [[ "$1" =~ ^(((file:(//)?)|((((git|ssh|http(s)?):(//)?)?)(([_a-zA-Z0-9]+@)?)([_a-zA-Z0-9.]+)((:[0-9]+)?)(:?)))([_a-zA-Z0-9./\-]+)(\.git)(/)?)?$ ]]
+  }
+
+  set +u
+  while [[ "$1" == "--"* ]]; do
+    opt="$1"
+    shift # expose next argument
+    case "$opt" in
+      "--") break 2 ;;
+      "--brew-git-remote")
+        HOMEBREW_BREW_GIT_REMOTE="$1"; shift ;;
+      "--brew-git-remote="*) # alternate format: --brew-git-remote=URL
+        HOMEBREW_BREW_GIT_REMOTE="${opt#*=}" ;;
+      "--core-git-remote")
+        HOMEBREW_CORE_GIT_REMOTE="$1"; shift ;;
+      "--core-git-remote="*) # alternate format: --core-git-remote=URL
+        HOMEBREW_CORE_GIT_REMOTE="${opt#*=}" ;;
+      *)
+        echo "Invalid option: ${opt}" >&2; exit 1 ;;
+    esac
+  done
+  if ! verify_git_remote_url "${HOMEBREW_BREW_GIT_REMOTE}"; then
+    echo "Invalid remote URL: HOMEBREW_BREW_GIT_REMOTE=${HOMEBREW_BREW_GIT_REMOTE}" >&2; exit 1
+  fi
+  if ! verify_git_remote_url "${HOMEBREW_CORE_GIT_REMOTE}"; then
+    echo "Invalid remote URL: HOMEBREW_CORE_GIT_REMOTE=${HOMEBREW_CORE_GIT_REMOTE}" >&2; exit 1
+  fi
+  set -u
+}
```

Comment:
I wonder if we could consider using just the first and second argument for these remotes unconditionally? CC @Homebrew/maintainers for any thoughts on all this.

Reply:
The user needs to pass two URLs as unconditional arguments. Even if he or she only wants to mirror one of the repo (`brew` or `brew-core`), not both of them.